### ISSUE TITLE
Hausdorff Loss

### ISSUE THREAD

**Issue by Dale-Black** [JuliaML/LossFunctions.jl#146]
I would like to add a simple Hausdorff Loss as proposed in (DOI: 10.1109/TMI.2019.2930068)

It approximates the Hausdorff distance to allow for direct minimization of the HD during training

```julia
function hd_loss(ŷ, y, ŷ_dtm, y_dtm)
    M = (ŷ .- y) .^ 2 .* (ŷ_dtm .^ 2 .+ y_dtm .^ 2)
    return loss = mean(M)
end
```

Should I open a PR for this?

**Comment by Dale-Black** [comment]
It might also be nice to add in a dice loss too even though this is already in Flux?

```julia
function dice_loss(ŷ, y)
    ϵ = 1e-5
    loss = 1 - ((2 * sum(ŷ .* y) + ϵ) / (sum(ŷ .* ŷ) + sum(y .* y) + ϵ))
end
```

**Comment by juliohm** [comment]
@Dale-Black can you please try to draft a PR following the current interface?

**Comment by Dale-Black** [comment]
Yes I would love to do that, but it might be 1.5 weeks once finals are over. Is there a way to have GitHub remind me about this in like 2 weeks??

**Comment by juliohm** [comment]
@Dale-Black I don't know of any feature in GitHub for reminders, but you can always set it in your own personal calendar.

We are cleaning up this repo once more, and it would be nice to have more contributors/maintainers. 👍🏽

**Comment by juliohm** [comment]
Closing as inactive. The Hausdorff distance is a distance between sets. I've implemented it in ImageDistances.jl in case someone needs a kickstart. Feel free to open a new issue if this feature is still needed, and we can see if it fits into the package scope.
